Source/panels/charpanel.cpp

```diff
@@ -0,0 +1,315 @@
+};
+
+struct panelEntry {
+	std::string label;
+	Displacement position;
+	int length;
+	Displacement labelOffset; // label's offset (end of the label vs the beginning of the stat box)
+	int labelLength;          // max label's length - used for line wrapping
+	int labelSpacing;
+	int statSpacing;
+	bool centered;
+	// must be set to true for stat boxes or they don't line up with the "spend stat" button
+	bool high;                                     // if the box should be using the 27px version or 26px
+	std::function<colorAndText()> statDisplayFunc; // function responsible for displaying stat
+};
+
+auto &myPlayer = Players[MyPlayerId];
+
+UiFlags GetBaseStatColor(CharacterAttribute attr)
+{
+	UiFlags style = UiFlags::ColorSilver;
+	if (myPlayer.GetBaseAttributeValue(attr) == myPlayer.GetMaximumAttributeValue(attr))
+		style = UiFlags::ColorGold;
+	return style;
+}
+
+UiFlags GetCurrentStatColor(CharacterAttribute attr)
+{
+	UiFlags style = UiFlags::ColorSilver;
+	int current = myPlayer.GetCurrentAttributeValue(attr);
+	int base = myPlayer.GetBaseAttributeValue(attr);
+	if (current > base)
+		style = UiFlags::ColorBlue;
+	if (current < base)
+		style = UiFlags::ColorRed;
+	return style;
+}
+
+UiFlags GetValueColor(int value, bool flip = false)
+{
+	UiFlags style = UiFlags::ColorSilver;
+	if (value > 0)
+		style = (flip ? UiFlags::ColorRed : UiFlags::ColorBlue);
+	if (value < 0)
+		style = (flip ? UiFlags::ColorBlue : UiFlags::ColorRed);
+	return style;
+}
+
+UiFlags GetMaxManaColor()
+{
+	return myPlayer._pMaxMana > myPlayer._pMaxManaBase ? UiFlags::ColorBlue : UiFlags::ColorSilver;
+}
+
+std::pair<int, int> GetDamage()
+{
+	int mindam = myPlayer._pIMinDam;
+	mindam += myPlayer._pIBonusDam * mindam / 100;
+	mindam += myPlayer._pIBonusDamMod;
+	if (myPlayer.InvBody[INVLOC_HAND_LEFT]._itype == ITYPE_BOW) {
+		if (myPlayer._pClass == HeroClass::Rogue)
+			mindam += myPlayer._pDamageMod;
+		else
+			mindam += myPlayer._pDamageMod / 2;
+	} else {
+		mindam += myPlayer._pDamageMod;
+	}
+	int maxdam = myPlayer._pIMaxDam;
+	maxdam += myPlayer._pIBonusDam * maxdam / 100;
+	maxdam += myPlayer._pIBonusDamMod;
+	if (myPlayer.InvBody[INVLOC_HAND_LEFT]._itype == ITYPE_BOW) {
+		if (myPlayer._pClass == HeroClass::Rogue)
+			maxdam += myPlayer._pDamageMod;
+		else
+			maxdam += myPlayer._pDamageMod / 2;
+	} else {
+		maxdam += myPlayer._pDamageMod;
+	}
+	return { mindam, maxdam };
+}
+
+colorAndText GetResistInfo(int8_t resist)
+{
+	UiFlags color = UiFlags::ColorBlue;
+	if (resist == 0)
+		color = UiFlags::ColorSilver;
+	else if (resist < 0)
+		color = UiFlags::ColorRed;
+	else if (resist >= MAXRESIST)
+		color = UiFlags::ColorGold;
+
+	return {
+		color, (resist >= MAXRESIST ? _("MAX") : fmt::format("{:d}%", resist))
+	};
+}
+
+panelEntry panelEntries[] = {
+	{ "", { 13, 14 }, 134, { 0, 0 }, 0, 0, 1, false, false,
+	    []() { return colorAndText { UiFlags::ColorSilver, myPlayer._pName }; } },
+	{ "level", { 57, 52 }, 45, { -3, 0 }, 0, 0, 1, false, false,

```

Comment:
```suggestion
	{ N_("Level"), { 57, 52 }, 45, { -3, 0 }, 0, 0, 1, false, false,
```

etc. for all the other labels.